### ISSUE TITLE
Bug in v2.11.0?

### ISSUE THREAD

**Issue by 54m4d** [rowrawer/stream-detector#182]
There is something confusing with the latest version, i'm not sure if i know how o describe it.

When you open a stream the indicated number flashes and disappears immediately, when you click the icon (current tab) there is nothing, switch to current session and you can see some links, switch back again to current tab, links are still there.

Now when you click away and click the icon again, you have to go through the above process, and the number indication is a hit and miss.

What gives?

btw, do have a IM messaging like telegram or something for faster discussions?

Thanks!

**Comment by rowrawer** [comment]
It works fine for me, but I think I know why you might be experiencing this. I'll try to push a fix, but Mozilla have been very slow in updating the listings lately.

>btw, do have a IM messaging like telegram or something for faster discussions?
No, this is just a hobby project. I only really spend time on it when I feel like it, so I'd rather not provide support this way.

**Comment by 54m4d** [comment]
It's the same on Chrome, maybe a default bug on all the browsers?

https://user-images.githubusercontent.com/52612348/218133441-2f091aa5-3057-46d9-9dd7-5da2483ef735.mp4

**Comment by rowrawer** [comment]
This is due to the sidebar implementation messing with the table. I'll have it fixed in the next update.

**Comment by 54m4d** [comment]
Good to know, thanks!

**Comment by Dean-Corso** [comment]
Hi @rowrawer,

thanks for the update but I also have to say your latest work is buggy and I also get that problem @54m4d is showing in the video and I have to switch from radiobutton to other radiobutton to get an updated view. So you told its the sidebar implementation problem what brings the question up.....what sidebar? I don't see any sidebar so where is it etc? Maybe you could explain next time what you did add for new features. I'am also still missing a option to clean all entries from all lists. Each time I start I have whole bunch of URLs into prev list if I have unchecked "ignore previous sessions on launch" and if I have checked that you just hide the radiobutton but the entries are still there. So my request in this case is, change this option to "clear all on launch or exit" and then clear also all entries completely. Its really pitty to see entries into prev sessions list which are pretty old. Just add a real clear option to make it straight you know. Just keep going. Thank you.

PS: The latest version I have test in Firefox. If this works fine then I install the Chrome version. Just as info.

**Comment by rowrawer** [comment]
>thanks for the update but I also have to say your latest work is buggy

It works fine for me, which is what I aim for.

>what sidebar? I don't see any sidebar so where is it etc?

As requested in https://github.com/rowrawer/stream-detector/issues/176.
How to open the sidebar: https://support.mozilla.org/en-US/kb/use-firefox-sidebar-access-bookmarks-history-synced
You can also assign a keyboard shortcut to open the sidebar or popup in the extensions list -> options -> manage extensions shortcuts.
It's up to the user to figure out how to use such features.

>if I have unchecked "ignore previous sessions on launch" and if I have checked that you just hide the radiobutton but the entries are still there

You can clear the list and then check the checkbox. The list should then remain empty. And even if it isn't, what does it matter if the tab isn't even there? I don't want to clutter the popup with yet another button to clear all lists as I don't really need it, considering there are only three tabs in total. You can just do it manually.

You've written about so many issues with the addon - may I suggest staying on an old version and disabling updates? You seem to have very particular needs for a free product and you don't really need to update if an old version is working fine for you.

**Comment by Dean-Corso** [comment]
Hi again,

strange that you don't get this problem like @54m4d did show in the video so I get the same problem (just did test Firefox). No entries are shown on first click on tool icon and then you need to switch to any other radiobutton and then the view gets updated and entries are shown.

Ah ok about sidebar now. I see it now and did set any shortcut to this option. Nice, but also here I don't get the entries to see just if I switch the radiobutton. Why don't you just add a button to show / hide the sidebar from options site or popup? I don't have any shortcuts free or otherwise I prefer using mouse click instead of keyboard. Just check this not showing entries problem on first click on icon or sidebar call. For you it works (did you test FF & Chrome?) and for @54m4d and me not.

Problem with that ignore option is that the entries are still there when I pick the All tabs radiobutton. Somehow the previouse sessions radiobutton ID seems to used with the "all tabs" radiobutton and does show all entries in there. There is no real clean clear function which works.

So I think thats the goal to show you the bugs / problems I found so that you can fix & improve them. Just depends on you whether you follow them or not. Sure I'am still using the old version too as I said before already few times but I would like to use just one only. Anyway, just don't uderstand why you don't like to hear my complaints so I'am trying to help you and thats all. I do not attack your work or make it bad etc you know.

**Comment by rowrawer** [comment]
>Problem with that ignore option is that the entries are still there when I pick the All tabs radiobutton. Somehow the previouse sessions radiobutton ID seems to used with the "all tabs" radiobutton and does show all entries in there. There is no real clean clear function which works.

Do you mean that the entries from the previous sessions are present in the all tabs section? That shouldn't be the case.

>So I think thats the goal to show you the bugs / problems I found so that you can fix & improve them. Just depends on you whether you follow them or not. Sure I'am still using the old version too as I said before already few times but I would like to use just one only. Anyway, just don't uderstand why you don't like to hear my complaints so I'am trying to help you and thats all. I do not attack your work or make it bad etc you know.

I'm just proposing a solution that would save you time - if I'm using software, I want to find something that works and fits my needs, and then spend as little time on it as possible. Maybe you have a different approach, that's fine.

**Comment by Dean-Corso** [comment]
Yes. Just try it out. Just clear all tabs manually first then check the ignore option and then get some entries in your list and now re-start the browser and then click on your app icon and switch to all tabs and there you can see all previous entries but this tab should be empty. So its working same like with unchecked ignore option just using the all tabs list instead. Maybe some ID value problem. Not sure whether you do use static ID numbers or not for the radiobutton resource. Just try it out.

Anyway, so I like your tool of course no question and I'am still using it and maybe I'am just a little pedantic in some thing. Sorry about that @rowrawer. Otherwise your tool is great and just keep going ok. 👍

**Comment by rowrawer** [comment]
>Yes. Just try it out. Just clear all tabs manually first then check the ignore option and then get some entries in your list and now re-start the browser and then click on your app icon and switch to all tabs and there you can see all previous entries but this tab should be empty. So its working same like with unchecked ignore option just using the all tabs list instead. Maybe some ID value problem. Not sure whether you do use static ID numbers or not for the radiobutton resource. Just try it out.

Interesting, I'll look into it.

**Comment by 54m4d** [comment]
Fixed in [1a84fa2](https://github.com/rowrawer/stream-detector/commit/1a84fa28b4622d594fe6b48ca48c0464824cd508)

Thanks!

**Comment by rowrawer** [comment]
>Yes. Just try it out. Just clear all tabs manually first then check the ignore option and then get some entries in your list and now re-start the browser and then click on your app icon and switch to all tabs and there you can see all previous entries but this tab should be empty. So its working same like with unchecked ignore option just using the all tabs list instead. Maybe some ID value problem. Not sure whether you do use static ID numbers or not for the radiobutton resource. Just try it out.

This should now be fixed in v2.11.2 once it gets approved by Mozilla.
